synapse/rest/admin/users.py

```diff
@@ -269,7 +269,10 @@ async def on_PUT(
                         target_user.to_string(), False, requester, by_admin=True
                     )
                 elif not deactivate and user["deactivated"]:
-                    if "password" not in body:
+                    if (
+                        "password" not in body
+                        and self.hs.config.password_localdb_enabled
```

Comment:
I'm not sure this will catch all the instances of this, would it make more sense to check if there's a mapping from SSO auth provider to this user?

In particular, I think many admins run with passwords completely disabled, and not specifically the localdb passwords disabled.

Reply:
IMO this not the best solution. But this is like `set_password` is implemented.
https://github.com/matrix-org/synapse/blob/1baab2035265cf2543fe3c0ef5412c1ac0740c7e/synapse/handlers/set_password.py#L37-L45

To improve it, we have also change the check in `set_password`.

Reply:
After further conversation in #synapse-dev:matrix.org, I suspect that this should check `self.hs.config.password_enabled and self.hs.config.password_localdb_enabled` since that is the only time it makes sense to set a password.